zip-0226.rst

```diff
@@ -325,6 +325,111 @@ A ZSA Asset Burn description is encoded in a transaction as an of a AssetBurn ty
+        ├── issue_actions_digest
+        └── issuanceValidatingKey
+
+In the specification below, nodes of the tree are presented in depth-first order.
+
+txid_digest
+-----------
+A BLAKE2b-256 hash of the following values ::
+
+   T.1: header_digest       (32-byte hash output)
+   T.2: transparent_digest  (32-byte hash output)
+   T.3: sapling_digest      (32-byte hash output)
+   T.4: orchard_digest      (32-byte hash output)
+   T.5: issuance_digest     (32-byte hash output)
+
+The personalization field remains the same as in ZIP 244 [#zip-0244]_. The branches ``T.1: header_digest``, ``T.2: transparent_digest``, and ``T.3: sapling_digest`` are as defined in ZIP 244.
+
+T.4: orchard_digest
+```````````````````
+When Orchard Actions are present in the transaction, this digest is a BLAKE2b-256 hash of the following values ::
+
+   T.4a: orchard_actions_compact_digest      (32-byte hash output)
+   T.4b: orchard_actions_assetid_digest      (32-byte hash output)
+   T.4c: orchard_actions_memos_digest        (32-byte hash output)
+   T.4d: orchard_actions_noncompact_digest   (32-byte hash output)
+   T.4e: flagsOrchard                        (1 byte)
+   T.4f: valueBalanceOrchard                 (64-bit signed little-endian)
+   T.4g: anchorOrchard                       (32 bytes)
+
+The value of ``T.4a: orchard_actions_compact_digest`` is the same as in ZIP 244 [#zip-0244]_. 
+
+T.4b: orchard_actions_assetid_digest
+''''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the subset of Zcash Shielded Asset identifier data for all Orchard Actions belonging to the transaction. For each Action, the following elements are included in the hash::
+
+   T.4b.i: assetid (field encoding bytes)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdOrcActAHash"
+
+T.4c: orchard_actions_memos_digest
+''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the subset of Orchard shielded memo field data for all Orchard
+Actions belonging to the transaction. For each Action, the following elements are included
+in the hash::
+
+    T.4c.i: encCiphertext[84..596] (contents of the encrypted memo field)
+
+The personalization field of this hash is defined identically to ZIP 244::
+
+  "ZTxIdOrcActMHash"
+
+
+T.4d: orchard_actions_noncompact_digest
+'''''''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the remaining subset of Orchard Action information **not** intended
+for inclusion in an updated version of the the ZIP 307 [#zip-0307]_ ``CompactBlock``
+format, for all Orchard Actions belonging to the transaction. For each Action,
+the following elements are included in the hash::
+
+   T.4d.i  : cv                    (field encoding bytes)
+   T.4d.ii : rk                    (field encoding bytes)
+   T.4d.iii: encCiphertext[596..]  (post-memo suffix of field encoding)
+   T.4d.iv : outCiphertext         (field encoding bytes)
+
+The personalization field of this hash is defined identically to ZIP 244::
+
+    "ZTxIdOrcActNHash"
```

Comment:
the personalizations remain unchanged from orchard:
```rust
const ZCASH_ORCHARD_ACTIONS_COMPACT_HASH_PERSONALIZATION: &[u8; 16] = b"ZTxIdOrcActCHash";
const ZCASH_ORCHARD_ACTIONS_MEMOS_HASH_PERSONALIZATION: &[u8; 16] = b"ZTxIdOrcActMHash";
const ZCASH_ORCHARD_ACTIONS_NONCOMPACT_HASH_PERSONALIZATION: &[u8; 16] = b"ZTxIdOrcActNHash";
```

and 

```rust
 ch.update(&action.nullifier().to_bytes());
        ch.update(&action.cmx().to_bytes());
        ch.update(&action.encrypted_note().epk_bytes);
        ch.update(&action.encrypted_note().enc_ciphertext[..84]); // TODO: make sure it is backward compatible with Orchard [..52]

        mh.update(&action.encrypted_note().enc_ciphertext[84..596]);

        nh.update(&action.cv_net().to_bytes());
        nh.update(&<[u8; 32]>::from(action.rk()));
        nh.update(&action.encrypted_note().enc_ciphertext[596..]);
        nh.update(&action.encrypted_note().out_ciphertext);
```

We just changed the length  (84,596)
Should fix the structure here and in orchard_actions_memos_digest, use bold for the changed values.

Reply:
Fixed the structure, but using bold to highlight that the range values have changed doesn't seem to be possible.